File: PopularShows/app/src/main/java/com/lgcode/popularshows/TvShowDetailActivity.java
package com.lgcode.popularshows;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.squareup.otto.Bus;

import glb.lg.popularshows.R;

import com.lgcode.popularshows.bus.BusProvider;
import com.lgcode.popularshows.models.TvShowDetailModel;
import com.lgcode.popularshows.presenters.TvShowDetailPresenter;
import com.lgcode.popularshows.views.TvShowDetailView;

/**
 * Created by le.garcia on 08/11/2016.
 */
public class TvShowDetailActivity extends AppCompatActivity {

    public static final String KEY_TV_SHOW = "tvShow";

    private Bus bus = BusProvider.getInstance();
    private TvShowDetailPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.detail_view);

        Intent intent = getIntent();

        TvShow tvShow = (TvShow) intent.getSerializableExtra(KEY_TV_SHOW);

        this.presenter = new TvShowDetailPresenter(
                new TvShowDetailModel(bus, tvShow),
                new TvShowDetailView(this, bus));
    }

    @Override
    protected void onStart() {
        super.onStart();
        bus.register(presenter);
    }

    @Override
    protected void onStop() {
        super.onStop();
        bus.unregister(presenter);
    }

}

File: KtPopularShows/app/src/main/java/example/lgcode/ktpopularshows/activity/TVShowDetailActivity.kt
package example.lgcode.ktpopularshows.activity

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import example.lgcode.ktpopularshows.R
import example.lgcode.ktpopularshows.domain.TVShow
import example.lgcode.ktpopularshows.mvp.presenter.TVShowDetailPresenter

class TVShowDetailActivity: AppCompatActivity() {

    private lateinit var presenter: TVShowDetailPresenter

    companion object {

        val KEY_TV_SHOW = "tvShow"

        fun getIntent(activity: Activity, tvShow: TVShow): Intent {
            val intent = Intent(activity, TVShowDetailActivity::class.java)
            intent.putExtra(KEY_TV_SHOW, tvShow)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_view)

        val tvShow = intent.extras[KEY_TV_SHOW] as TVShow
        presenter = TVShowDetailPresenter()
        presenter.init(this)
    }

}